crates/aptos/src/governance/mod.rs

```diff
@@ -183,3 +203,166 @@ impl std::fmt::Display for ProposalMetadata {
         )
     }
 }
+
+#[derive(Clone, Debug, Default, Serialize)]
+pub struct ScriptHash {
+    hash: String,
+    bytecode: String,
+}
+
+#[derive(Parser)]
+pub struct PrepareScript {
+    /// Path to the Move package that contains the execution script
+    #[clap(long, parse(from_os_str))]
+    pub path: PathBuf,
+}
+
+#[async_trait]
+impl CliCommand<ScriptHash> for PrepareScript {
+    fn command_name(&self) -> &'static str {
+        "PrepareScript"
+    }
+
+    async fn execute(mut self) -> CliTypedResult<ScriptHash> {
+        let build_config = BuildConfig {
+            additional_named_addresses: BTreeMap::new(),
+            generate_abis: false,
+            generate_docs: false,
+            ..Default::default()
+        };
+
+        let compiled_package = compile_move(build_config, self.path.as_path())?;
+        let scripts_count = compiled_package.scripts().count();
+
+        if scripts_count != 1 {
+            return Err(CliError::UnexpectedError(format!(
+                "Prepare 1 script a time. Found {} scripts.",
+                scripts_count
+            )));
+        }
+
+        let script = *compiled_package
+            .scripts()
+            .collect::<Vec<_>>()
+            .get(0)
+            .unwrap();
+
+        match script.unit {
+            CompiledUnitEnum::Script(ref s) => {
+                let mut bytes = vec![];
+
+                s.script.serialize(&mut bytes).map_err(|err| {
+                    CliError::UnexpectedError(format!("Unexpected error: {}", err))
+                })?;
+
+                Ok(ScriptHash {
+                    hash: hex::encode(HashValue::sha3_256_of(bytes.as_slice()).to_vec()),
+                    bytecode: hex::encode(bytes),
+                })
+            }
+            _ => Err(CliError::UnexpectedError("Unkown error".to_string())),
+        }
+    }
+}
+
+#[derive(Parser)]
+pub struct RunScript {
+    #[clap(flatten)]
+    pub(crate) txn_options: TransactionOptions,
+
+    /// Path to the compiled script file
+    #[clap(long, parse(from_os_str))]
+    pub path: PathBuf,
+
+    /// Hex encoded arguments separated by spaces.
+    ///
+    /// Example: `0x01 0x02 0x03`
+    #[clap(long, multiple_values = true)]
+    pub(crate) args: Vec<ArgWithType>,
```

Comment:
This is sorta surprising to me, I assumed that the compiled script couldn't take in arguments. Like folks vote on a proposal, but then that proposal could do different things depending on the argument? Pretty strange. cc @movekevin

Reply:
It takes parameters